yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnShuffleEncryptionSuite.scala

```diff
@@ -0,0 +1,289 @@
+import org.apache.spark.serializer.{KryoSerializer, SerializerInstance}
+import org.apache.spark.shuffle.{RecordingManagedBuffer, BaseShuffleHandle,
+  BlockStoreShuffleReader, IndexShuffleBlockResolver}
+import org.apache.spark.shuffle.sort.{SerializedShuffleHandle, UnsafeShuffleWriter}
+import org.apache.spark.storage._
+import org.apache.spark.util.Utils
+
+private[spark] class YarnShuffleEncryptionSuite extends SparkFunSuite with Matchers with
+                                                        BeforeAndAfterEach with LocalSparkContext {
+
+  @Mock(answer = RETURNS_SMART_NULLS) private[this] var blockManager: BlockManager = _
+  @Mock(answer = RETURNS_SMART_NULLS) private[this] var diskBlockManager: DiskBlockManager = _
+  @Mock(answer = RETURNS_SMART_NULLS) private[this] var taskContext: TaskContext = _
+  @Mock(answer = RETURNS_SMART_NULLS) private[this] var blockResolver: IndexShuffleBlockResolver = _
+  @Mock(
+    answer = RETURNS_SMART_NULLS) private[this] var shuffleDep: ShuffleDependency[Int, Int, Int] = _
+
+  private[this] val NUM_PARTITITONS = 4
+  private[this] val NUM_MAPS = 1
+  private[this] val SHUFFLE_ID = 0
+  private[this] val REDUCE_ID = 1
+  private[this] val conf = new SparkConf()
+  private[this] val memoryManager = new TestMemoryManager(conf)
+  private[this] val taskMemoryManager = new TaskMemoryManager(memoryManager, 0)
+  private[this] val serializer = new KryoSerializer(conf)
+  private[this] val hashPartitioner = new HashPartitioner(NUM_PARTITITONS)
+  private[this] val taskMetrics = new TaskMetrics()
+  private[this] val spillFilesCreated = new java.util.LinkedList[File]()
+
+  private[this] var tempDir: File = _
+  private[this] var mergedOutputFile: File = _
+  private[this] var partitionSizesInMergedFile: Array[Long] = _
+  private[this] val ugi = UserGroupInformation.createUserForTesting("testuser", Array[String] {
+    "testgroup"
+  })
+
+  // Create a mocked shuffle handle to pass into HashShuffleReader.
+  private[this] val shuffleHandle = {
+    val dependency = mock(classOf[ShuffleDependency[Int, Int, Int]])
+    when(dependency.serializer).thenReturn(Some(serializer))
+    when(dependency.aggregator).thenReturn(None)
+    when(dependency.keyOrdering).thenReturn(None)
+    new BaseShuffleHandle(SHUFFLE_ID, NUM_MAPS, dependency)
+  }
+
+  // Make a mocked MapOutputTracker for the shuffle reader to use to determine what
+  // shuffle data to read.
+  private[this] val mapOutputTracker = mock(classOf[MapOutputTracker])
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    MockitoAnnotations.initMocks(this)
+    tempDir = Utils.createTempDir("test", "test")
+    mergedOutputFile = File.createTempFile("mergedoutput", "", tempDir)
+    System.setProperty("SPARK_YARN_MODE", "true")
+    // Create a SparkContext as a convenient way of setting SparkEnv (needed because some of the
+    // shuffle code calls SparkEnv.get()).
+    sc = new SparkContext("local", "test", conf)
```

Comment:
Can you mock the SparkEnv (and call SparkEnv.set) instead? That would be cheaper than instantiating a SparkContext, but no big deal.